libs/pika/async_mpi/tests/unit/pool_creation.cpp

```diff
@@ -0,0 +1,138 @@
+{
+    int size, rank;
+    MPI_Comm comm = MPI_COMM_WORLD;
+    MPI_Comm_size(comm, &size);
+    MPI_Comm_rank(comm, &rank);
+
+    MPI_Datatype datatype = MPI_INT;
+
+    {
+        // Register polling on our custom pool
+        mpi::enable_polling enable_polling(mpi::exception_mode::install_handler, pool_name);
+        // Success path
+        {
+            // MPI function pointer
+            int data = 0, count = 1;
+            if (rank == 0) { data = 42; }
+            auto s = mpi::transform_mpi(ex::just(&data, count, datatype, 0, comm), MPI_Ibcast);
+            tt::sync_wait(PIKA_MOVE(s));
+            PIKA_TEST_EQ(data, 42);
+        }
+    }    // let the polling go out of scope
+
+    {
+        // Register polling on default pool
+        mpi::enable_polling enable_polling(mpi::exception_mode::install_handler,
+            pika::resource::get_partitioner().get_default_pool_name());
+        // Success path
+        {
+            // MPI function pointer
+            int data = 0, count = 1;
+            if (rank == 0) { data = 42; }
+            auto s = mpi::transform_mpi(ex::just(&data, count, datatype, 0, comm), MPI_Ibcast);
+            tt::sync_wait(PIKA_MOVE(s));
+            PIKA_TEST_EQ(data, 42);
+        }
+    }    // let the polling go out of scope
+
+    pika::finalize();
+    return EXIT_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+void init_resource_partitioner_manual(
+    pika::resource::partitioner& rp, pika::program_options::variables_map const&)
+{
+    // Disable idle backoff on our custom pool
+    using pika::threads::scheduler_mode;
+    auto mode = scheduler_mode::default_mode;
+    mode = scheduler_mode(mode & ~scheduler_mode::enable_idle_backoff);
+
+    // Create a thread pool with a single core that we will use for communication related tasks
+    rp.create_thread_pool(
+        random_name1, pika::resource::scheduling_policy::local_priority_fifo, mode);
+    rp.add_resource(rp.sockets()[0].cores()[0].pus()[0], random_name1);
+}
+
+//----------------------------------------------------------------------------
+void init_resource_partitioner_mpi(
+    pika::resource::partitioner& rp, pika::program_options::variables_map const&)
+{
+    pika::mpi::experimental::detail::create_pool(
+        rp, random_name2, mpi::polling_pool_creation_mode::mode_force_create);
+}
+
+//----------------------------------------------------------------------------
+int main(int argc, char* argv[])
+{
+    MPI_Init(&argc, &argv);
+
+    int result = 0;
+    {
+        // ---------------------------------------
+        // Start runtime and collect runtime exit status, turn on auto pool creation
+        pika::init_params init_args;
+        init_args.cfg.emplace_back("pika.mpi.enable_pool=" + std::to_string(true));
+        result |= pika::init([&]() { return pika_main(""); }, argc, argv, init_args);
+        PIKA_TEST_EQ(result, 0);
+    }
```

Comment:
Just for you to know: you don't have to accumulate and return the `result` from `main`. If any of the `PIKA_TEST_EQ(result, 0)` fail, the test will exit with a non-zero code anyway. Since you're testing each `result` returned from `pika::init`, I think you can remove the accumulation and just define `int result = 0` in each scope.